st2api/st2api/controllers/v1/auth.py

```diff
@@ -97,11 +101,27 @@ def get_all(self, requester_user, show_secrets=None):
         """
         mask_secrets = self._get_mask_secrets(show_secrets=show_secrets,
                                               requester_user=requester_user)
-        api_key_dbs = ApiKey.get_all()
-        api_keys = [ApiKeyAPI.from_model(api_key_db, mask_secrets=mask_secrets)
-                    for api_key_db in api_key_dbs]
 
-        return api_keys
+        if limit and int(limit) > self.max_limit:
+            msg = 'Limit "%s" specified, maximum value is "%s"' % (limit, self.max_limit)
+            raise ValueError(msg)
+
+        api_key_dbs = ApiKey.get_all(limit=limit, offset=offset)
+
+        try:
+            api_keys = [ApiKeyAPI.from_model(api_key_db, mask_secrets=mask_secrets)
+                        for api_key_db in api_key_dbs]
+        except OverflowError:
```

Comment:
@Kami: Good point. Apikeys endpoint use https://github.com/StackStorm/st2/blob/master/st2common/st2common/persistence/auth.py#L84. Not sure if we can refactor here. However, all other endpoints were not even addressing OverflowError, fixed here: https://github.com/StackStorm/st2/pull/3486/commits/489fa2d357bbd339f887f69da2d7e4df798214dc. Also, some endpoints on openapi.yaml didn't have offset as a parameter.